packages/common/src/common.ts

```diff
@@ -75,6 +73,8 @@ export class Common {
         (this._chainParams.customHardforks && this._chainParams.customHardforks[hf.name]),
     ])
     this._hardfork = this.DEFAULT_HARDFORK
+    this._params = { ...(opts.params ?? {}) } // copy
```

Comment:
If the shallow copy should be removed, then this one should likely also be addressed :smile:

Reply:
Oh yes, good point! Thanks!